### ISSUE TITLE
Bot Freezes when MouseOver locks the Overview with Depleted Asteroid

### ISSUE THREAD

**Issue by zoom-maestro** [Viir/bots#33]
![image](https://user-images.githubusercontent.com/54839586/111864443-32d86a00-891e-11eb-9e10-7176fb9b6836.png)

I would like to suggest the mouse moves to a "home spot" away from the Overview window after the bot detects that there is no longer a target. This will move the mouse off the Overview and avoid it trying to retarget an asteroid that no longer exists (and is no longer targetable)

As a side note, after an asteroid no longer exists, all miners should be powered down. This will avoid the wasted time and cycle. This moving of the mouse can also serve as the Overview reset/unlocking needed in the paragraph before this one.

To summarize:

- If no Target detected,
-- Mouseover and tap all the mining lasers to disable any active ones and to get ready any inactive ones.
---- This action will keep Overview from locking and allow depleted asteroid to be removed from Overview.
-- Then continue (as usual) and Approach/Target the next asteroid. 
-- Lasers that were tapped should auto-activate on the new target.
--- Existing bot-checks will see this and bypass activation.
--- Existing bot-checks will also see any off lasers and turn them on.

(Two birds with one stone)

**Comment by Viir** [comment]
Thank you for sharing your experience!

It looks like it is specific to EVE Online, so I will add a tag for that so that other people can find it better.

Judging by the screenshot, you could also avoid the entry representing the vanished object. That should be sufficient to achieve this behavior you described above:

> [...] avoid it trying to retarget an asteroid that no longer exists

That being said, let's continue with the approach you described. Turning off the miners should be simple enough. For example, we could use app 5e373c6bfb as a base to start from: https://github.com/Viir/bots/tree/6c184f822c1c0ebee0acfc00beb4a5e2faf28cdc/implement/applications/eve-online/eve-online-mining-bot

In that app, we already have this function to click a module: https://github.com/Viir/bots/blob/6c184f822c1c0ebee0acfc00beb4a5e2faf28cdc/implement/applications/eve-online/eve-online-mining-bot/BotEngineApp.elm#L461

Regarding the condition: There is also already this code to branch on if there is a target:
https://github.com/Viir/bots/blob/6c184f822c1c0ebee0acfc00beb4a5e2faf28cdc/implement/applications/eve-online/eve-online-mining-bot/BotEngineApp.elm#L440-L443

As far as I understood your post, we can insert the additional activity in that location and fallback to the previous behavior if there are no mining modules active anymore.

We can reuse this existing block. We just have to remove the negation in the filter to take only active modules:
https://github.com/Viir/bots/blob/6c184f822c1c0ebee0acfc00beb4a5e2faf28cdc/implement/applications/eve-online/eve-online-mining-bot/BotEngineApp.elm#L452-L462

Here is a complete implementation:
https://github.com/Viir/bots/commit/3e8b1c54bdc42bfe49ce2d3647e2ec9d0e3d98fe

**Comment by zoom-maestro** [comment]
Upon further testing and observation. I've found that the exception is being caused not by the current running of bot. But, actually went the bot is restarted after its 15 minute termination. Although, I do think the exception needs to be looked into, the simple avoidance is to use an "online session" to ensure that the bot is continuous and doesn't need to be restarted.

That being said, upon restarting the bot after the 15 minute termination, the bot will lock the overview if gameplay is in commencement. This means the player is out in space and in the mid-process (not docked in a station). When the target asteroid is depleted (from the already commenced gameplay), the overview is already locked at that point from the restarting the bot app.

This is what I observed upon restarting the bot after it terminated. 
![image](https://user-images.githubusercontent.com/54839586/111916260-40354780-8a37-11eb-9470-787530240aa7.png)

At this point the bot is in a waiting. For what state it is waiting for, I'm not sure. Probably multiple states? Most likely for the asteroid to be depleted or for the ore hold to reach its capacity? Ether way, when the asteroid depletes, the "ghost" presence of it stays in the Overview. Unfortunately this "ghost" cannot be targeted and this is where an exception needs to be made.

![image](https://user-images.githubusercontent.com/54839586/111916391-fa2cb380-8a37-11eb-915d-204a83e2e2ea.png)

The lack of target needs to be detected and the "ghost" overview item also needs to be noticed. The action: select and target the next valid target item in the Overview.

**Comment by zoom-maestro** [comment]
Purchased Credits to test theory behind continuous bot running and restarting the bot. If the issue still persists, I will dig deeper into the resolution.

**Comment by zoom-maestro** [comment]
Performing Online Sessions has reduced this issue considerably. However infrequent, it still occurs on occasion. 

On a side note, the fix in this fork is not working either way. I think the better approach will be to detect the lack of process, determine the "color" (or status) of the first line somehow and target the 2nd line instead.

The deactivation of the strip miner is causing a separate issue. It seems to be disabling the wrong miner and leaving the one that should be deactivated as activated. The is causing the capacitor to deplete and subsequently causing more issues and not solving any. I would ask that this solution be dissolved.

**Comment by zoom-maestro** [comment]
This solution does not work and may cause more issues than it fixes. Can we throw away this implementation?

**Comment by Viir** [comment]
> Can we throw away this implementation?

Yes, thank you for the clarification and sharing your results.
